### ISSUE TITLE
Improve error message for 'if (const x = 10) {}'

### ISSUE THREAD

**Issue by singalhimanshu** [quick-lint/quick-lint-js#424]
quick-lint-js reports crappy error message for the following code:
```js
if (const x = 10) {}
```
Current error message:
```sh
main.js:5:6: error: if statement is missing ')' around condition [E018]
main.js:5:6: error: missing body for 'if' statement [E150]
main.js:5:6: note: a lexical declaration is not allowed as the body of an 'if' statement [E150]
main.js:5:18: error: missing semicolon after statement [E027]
main.js:5:18: error: unmatched parenthesis [E056]
main.js:5:19: error: missing semicolon after statement [E027]
```
